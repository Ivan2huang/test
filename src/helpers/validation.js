export const required = value => !value;

export const email = value =>
  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value);

export const password = value =>
  !/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$/.test(value);

export const otp = value => !/\d{6}/.test(value);

export const maxLength = (max, value) => value && value.length > max;
