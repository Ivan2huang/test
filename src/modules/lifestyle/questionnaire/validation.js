import { QUESTIONNAIRE_TYPES, BMI_UNIT } from './constants';

const {
  height: { CM, IN, FTIN },
  weight: { KG, LB },
} = BMI_UNIT;

const HEIGHT = {
  [CM]: {
    min: 50,
    max: 300,
  },
  [FTIN]: {
    min: 3,
    max: 10,
  },
  [IN]: {
    min: 0,
    max: 11,
  },
};

const WEIGHT = {
  [KG]: {
    min: 30,
    max: 650,
  },
  [LB]: {
    min: 50,
    max: 400,
  },
};

const WAIST = {
  [CM]: {
    min: 40,
    max: 300,
  },
  [IN]: {
    min: 20,
    max: 200,
  },
};

// eslint-disable-next-line import/prefer-default-export
export const validate = (values, props) => {
  const { aboutMe = {} } = values;
  const {
    heightOne,
    heightTwo,
    heightUnit,
    weight,
    weightUnit,
    waistCircumference,
    waistUnit,
  } = aboutMe;
  const { choicesQuestions, healthQuestions } = props;
  const errors = {
    [QUESTIONNAIRE_TYPES.aboutMe]: {},
    [QUESTIONNAIRE_TYPES.choices]: {},
    [QUESTIONNAIRE_TYPES.health]: {},
  };
  const { min: minHeight, max: maxHeight } = HEIGHT[heightUnit] || {};
  const { min: minWeight, max: maxWeight } = WEIGHT[weightUnit] || {};
  const { min: minWaist, max: maxWaist } = WAIST[waistUnit] || {};

  if (!heightOne || heightOne < minHeight || heightOne > maxHeight) {
    errors.aboutMe.heightOne = {
      _error: true,
    };
  }
  if (heightUnit === FTIN) {
    const { min: minHeightTwo, max: maxHeightTwo } = HEIGHT[IN];
    if (!heightTwo || heightTwo < minHeightTwo || heightTwo > maxHeightTwo) {
      errors.aboutMe.heightTwo = {
        _error: true,
      };
    }
  }

  if (!weight || weight < minWeight || weight > maxWeight) {
    errors.aboutMe.weight = {
      _error: true,
    };
  }

  if (
    waistCircumference &&
    (waistCircumference < minWaist || waistCircumference > maxWaist)
  ) {
    errors.aboutMe.waistCircumference = {
      _error: true,
    };
  }

  const questionsValidation = (type, questions, ignored = []) => {
    const typeValues = values[type] || {};
    const typeErrors = errors[type];
    return questions.forEach(({ name }) => {
      if (!ignored.includes(name)) {
        const value = typeValues[name];
        if (
          (value !== 0 && !value) ||
          (Array.isArray(value) && value.length === 0)
        ) {
          typeErrors[name] = {
            _error: true,
          };
        }
      }
    });
  };

  questionsValidation(QUESTIONNAIRE_TYPES.choices, choicesQuestions);
  questionsValidation(QUESTIONNAIRE_TYPES.health, healthQuestions);

  return errors;
};
