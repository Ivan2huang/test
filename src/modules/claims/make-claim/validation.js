import { formatAmount } from '../../../helpers/helpers';

const getAmount = value => (value ? parseFloat(value.replace(',', '')) : 0);

export const validateAnotherInsurerAmount = (value, values) => {
  if (+value === 0) {
    return {
      id: 'claims.makeClaimForm.validations.greaterThanZero.otherInsurerAmount',
      defaultMessage: 'Amount should be greater than 0',
    };
  }
  if (getAmount(value) >= getAmount(values.claim.receiptAmount)) {
    return {
      id: 'claims.makeClaimForm.otherInsurerAmount.helperText',
      defaultMessage: 'Must be lower than receipt amount',
    };
  }

  return '';
};

export const validateReceiptAmount = (value, _, props) => {
  const { maxReceiptAmount, intl } = props;
  if (+value === 0) {
    return {
      id: 'claims.makeClaimForm.validations.greaterThanZero.receiptAmount',
      defaultMessage: 'Amount should be greater than 0',
    };
  }
  if (getAmount(value) > maxReceiptAmount) {
    const formattedAmount = formatAmount(intl, maxReceiptAmount);
    return {
      id: 'claims.makeClaimForm.validations.max.receiptAmount',
      defaultMessage: `For amounts above HK$ ${formattedAmount} submit a physical claim forms`,
      values: {
        maxReceiptAmount: formattedAmount,
      },
    };
  }

  return '';
};

export const validate = (values, props) => {
  const errors = {};

  if (values.receipts.files.length === 0) {
    errors.receipts = {};
    errors.receipts.files = {
      _error: true,
    };
  }

  if (props.referralRequired && values.referral.files.length === 0) {
    errors.referral = {};
    errors.referral.files = {
      _error: true,
    };
  }

  if (
    props.anotherInsurerEnabled &&
    values.settlementAdvices.files.length === 0
  ) {
    errors.settlementAdvices = {};
    errors.settlementAdvices.files = {
      _error: true,
    };
  }

  if (props.isChineseHerbalist && values.prescriptions.files.length === 0) {
    errors.prescriptions = {};
    errors.prescriptions.files = {
      _error: true,
    };
  }

  return errors;
};
