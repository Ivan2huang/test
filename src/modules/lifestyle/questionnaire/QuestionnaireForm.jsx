import React from 'react';
import { FormSection, reduxForm } from 'redux-form';
import * as PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import { Box, Card, Divider, withStyles } from '@material-ui/core';

import AboutMe from './AboutMeContainer';
import QuestionsSection from './QuestionsSection';
import { validate } from './validation';
import { formatMessage } from '../../../helpers/helpers';
import { QUESTIONNAIRE_TYPES, QUESTIONNAIRE_FORM } from './constants';
import Images from '../../../constants/images';
import { FutureMeContainer } from './future-me';
import TrackingButton from '../../../uiComponents/TrackingButton';
import { CATEGORIES } from '../../../constants/analytics';
import Typography from '../../../uiComponents/Typography';

import FaceAgingErrorModal from './FaceAgingErrorModal';

const styles = theme => ({
  cardRoot: {
    backgroundColor: theme.background,
    boxShadow: 'none',
  },
});

const QuestionnaireForm = ({
  intl,
  choicesQuestions,
  healthQuestions,
  faceAgingImageError,
  toggleErrorModal,
  handleSubmit,
  change,
  classes,
}) => (
  <>
    <Card classes={{ root: classes.cardRoot }}>
      <form onSubmit={handleSubmit} autoComplete="off">
        <FormSection name="aboutMe">
          <AboutMe fieldChange={change} />
        </FormSection>
        <Divider />
        <FormSection name={QUESTIONNAIRE_TYPES.choices}>
          <QuestionsSection
            title={formatMessage(
              intl,
              'lifestyle.questionnaire.myChoices.header',
              'My choices',
            )}
            imageUrl={Images.LIFESTYLE_QUESTIONNAIRE_MY_CHOICES}
            questions={choicesQuestions}
          />
        </FormSection>
        <Divider />
        <FormSection name={QUESTIONNAIRE_TYPES.health}>
          <QuestionsSection
            title={formatMessage(
              intl,
              'lifestyle.questionnaire.myHealth.header',
              'My health',
            )}
            imageUrl={Images.LIFESTYLE_QUESTIONNAIRE_MY_HEALTH}
            questions={healthQuestions}
          />
        </FormSection>
        <FutureMeContainer fieldChange={change} />
        <Divider />
        <Box display="flex" justifyContent="center" pt={11} pb={16}>
          <TrackingButton
            data-testid="btn-submit-questionnaire"
            type="submit"
            color="primary"
            variant="contained"
            trackingData={{
              category: CATEGORIES.HEALTH_QUESTIONNAIRE,
              action: 'Submit health questionaire',
            }}
          >
            {formatMessage(
              intl,
              'lifestyle.questionnaire.submitButton.label',
              'Show results',
            )}
          </TrackingButton>
        </Box>
      </form>
      <FaceAgingErrorModal
        open={faceAgingImageError}
        onClose={() => toggleErrorModal(false)}
      />
    </Card>
    <Box mt={{ xs: 0, md: 10 }}>
      <Typography type="style-8" color="highEmphasis">
        {formatMessage(
          intl,
          'lifestyle.questionnaire.disclaimerTitle',
          'Disclaimer',
        )}
      </Typography>
      <Typography type="style-8" color="mediumEmphasis">
        {formatMessage(
          intl,
          'lifestyle.questionnaire.disclaimerMessage',
          'This questionnaire is designed to provide you with an indication of your lifestyle and its impact on your health/wellness. It is NOT intended to constitute professional medical advice, diagnosis or treatment, and is NOT a substitute for a medical health assessment by a qualified medical professional. If youâ€™re experiencing signs and symptoms we recommend you seek medical attention from a qualified medical professional. By submitting this questionnaire, you do so in at your own choice and in agreement with this disclaimer.',
        )}
      </Typography>
    </Box>
  </>
);

const questionPropTypes = PropTypes.arrayOf(
  PropTypes.exact({
    questionOrder: PropTypes.string.isRequired,
    questionId: PropTypes.string.isRequired,
    dependentQuestion: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    section: PropTypes.string,
    options: PropTypes.arrayOf(
      PropTypes.exact({
        label: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired,
        hasInversion: PropTypes.bool,
      }),
    ).isRequired,
    activationAnswer: PropTypes.arrayOf(PropTypes.string),
    hasDefaultValue: PropTypes.bool,
    parentQuestionId: PropTypes.string,
    titleLabel: PropTypes.string,
    selectFieldLabel: PropTypes.string,
    buttonLabel: PropTypes.string,
  }).isRequired,
);

QuestionnaireForm.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.shape({}).isRequired,
  choicesQuestions: questionPropTypes.isRequired,
  healthQuestions: questionPropTypes.isRequired,
  faceAgingImageError: PropTypes.bool.isRequired,
  toggleErrorModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  change: PropTypes.func.isRequired,
};

const onSubmitFail = errors => {
  const {
    [QUESTIONNAIRE_TYPES.aboutMe]: aboutMe,
    [QUESTIONNAIRE_TYPES.choices]: choices,
    [QUESTIONNAIRE_TYPES.health]: health,
  } = errors;
  const flattenErrors = {
    ...aboutMe,
    ...choices,
    ...health,
  };
  const firstErrorId = Object.keys(flattenErrors)[0];
  const firstErrorElement = document.querySelector(`[data-id=${firstErrorId}]`);

  firstErrorElement.scrollIntoView(true);
};

const getFilteredValues = (type, questions, values) => {
  return questions.reduce((acc, cur) => {
    const key = cur.name;
    acc[key] = values[QUESTIONNAIRE_TYPES[type]][key];
    return acc;
  }, {});
};

const onSubmit = (values, _, props) => {
  const { choicesQuestions, healthQuestions, futureMeImageChanged } = props;
  props.submitQuestionnaire({
    ...values,
    [QUESTIONNAIRE_TYPES.choices]: getFilteredValues(
      QUESTIONNAIRE_TYPES.choices,
      choicesQuestions,
      values,
    ),
    [QUESTIONNAIRE_TYPES.health]: getFilteredValues(
      QUESTIONNAIRE_TYPES.health,
      healthQuestions,
      values,
    ),
    isFaceAgingImageChanged: futureMeImageChanged,
  });
};

const QuestionnaireFormWrapper = reduxForm({
  validate,
  onSubmitFail,
  onSubmit,
  form: QUESTIONNAIRE_FORM,
  enableReinitialize: true,
})(QuestionnaireForm);

QuestionnaireFormWrapper.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  choicesQuestions: PropTypes.arrayOf(
    PropTypes.exact({
      questionOrder: PropTypes.string.isRequired,
      questionId: PropTypes.string.isRequired,
      dependentQuestion: PropTypes.bool.isRequired,
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      section: PropTypes.string,
      options: PropTypes.arrayOf(
        PropTypes.exact({
          label: PropTypes.string.isRequired,
          value: PropTypes.string.isRequired,
          hasInversion: PropTypes.bool,
        }),
      ).isRequired,
      activationAnswer: PropTypes.arrayOf(PropTypes.string),
      hasDefaultValue: PropTypes.bool,
      parentQuestionId: PropTypes.string,
      titleLabel: PropTypes.string,
      selectFieldLabel: PropTypes.string,
      buttonLabel: PropTypes.string,
    }).isRequired,
  ).isRequired,
  submitQuestionnaire: PropTypes.func.isRequired,
};

export default injectIntl(withStyles(styles)(QuestionnaireFormWrapper));
