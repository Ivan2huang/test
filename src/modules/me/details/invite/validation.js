import moment from 'moment';
import { formatMessage } from '../../../../helpers/helpers';
import {
  IsCategorySpouse,
  IsCategoryChild,
} from '../../../../helpers/relationships';
import { email } from '../../../../helpers/validation';
import CONFIG from '../../../../constants/config';

export const hasValue = value => value && (value.trim ? value.trim() : true);

export const required = value => hasValue(value);

export const validateEmail = (value, _, props) => {
  const { intl } = props;
  const notValidEmail = email;

  if (notValidEmail(value)) {
    return formatMessage(
      intl,
      'me.tabs.myDetails.invite.dependent.invalidEmail',
      'Enter a valid email address',
    );
  }
  return '';
};

export const validateRequired = (value, _, props) => {
  const { intl } = props;

  if (!required(value)) {
    return formatMessage(
      intl,
      'me.tabs.myDetails.invite.dependent.isRequired',
      'Required',
    );
  }
  return '';
};

export const validateEmailsMatch = (intl, dependentEmail, confirmEmail) => {
  if (dependentEmail.toLowerCase() !== confirmEmail.toLowerCase()) {
    return formatMessage(
      intl,
      'me.tabs.myDetails.invite.dependent.isEmailNotMatched',
      'Emails do not match',
    );
  }

  return '';
};

const getDaysFromDate = date => {
  const now = moment();
  return moment.duration(now.diff(date)).asDays();
};

const getDateYearsAgo = years => {
  return moment()
    .subtract(years, 'years')
    .endOf('day');
};

const validateSpouseAge = (intl, dobDate) => {
  const { minSpouseAge } = CONFIG;
  const ageInDays = getDaysFromDate(dobDate);
  const minAgeDays = getDaysFromDate(getDateYearsAgo(minSpouseAge)) + 1;
  const isValid = ageInDays > minAgeDays;

  if (!isValid) {
    return formatMessage(
      intl,
      'me.tabs.myDetails.invite.dependent.invalidSpouseAge',
      'Your spouse does not meet the age requirement to access the dependent app and website for HSBC Benefits',
      { minSpouseAge },
    );
  }

  return '';
};

const validateChildAge = (intl, dobDate) => {
  const { minChildAge, maxChildAge } = CONFIG;
  const ageInDays = getDaysFromDate(dobDate);
  const minAgeDays = getDaysFromDate(getDateYearsAgo(minChildAge)) + 1;
  const maxAgeDays = getDaysFromDate(getDateYearsAgo(maxChildAge));
  const isValid = ageInDays > minAgeDays && ageInDays < maxAgeDays;

  if (!isValid) {
    return formatMessage(
      intl,
      'me.tabs.myDetails.invite.dependent.invalidChildAge',
      'Your child does not meet the age requirement to access the dependent app and website for HSBC Benefits',
      { minChildAge, maxChildAge },
    );
  }

  return '';
};

export const validateDateOfBirth = (
  intl,
  dateOfBirth,
  relationshipCategory,
) => {
  if (IsCategorySpouse(relationshipCategory)) {
    return validateSpouseAge(intl, dateOfBirth);
  }
  if (IsCategoryChild(relationshipCategory)) {
    return validateChildAge(intl, dateOfBirth);
  }

  return '';
};
