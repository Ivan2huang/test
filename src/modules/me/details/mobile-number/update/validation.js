import { required } from '../../../../../helpers/validation';
import { formatMessage } from '../../../../../helpers/helpers';

const validate = (values, { intl }) => {
  const errors = {};
  const {
    newMobileNumberCountryCode,
    newMobileNumber,
    confirmNewMobileNumberCountryCode,
    confirmNewMobileNumber,
  } = values;

  if (required(newMobileNumber) || required(newMobileNumberCountryCode)) {
    errors.newMobileNumber = formatMessage(
      intl,
      'validations.invalid.phoneNumber',
      'Enter valid mobile number',
    );
  } else if (
    required(confirmNewMobileNumber) ||
    required(confirmNewMobileNumberCountryCode)
  ) {
    errors.confirmNewMobileNumber = formatMessage(
      intl,
      'validations.invalid.phoneNumber',
      'Enter valid mobile number',
    );
  } else if (newMobileNumber !== confirmNewMobileNumber) {
    errors.confirmNewMobileNumber = formatMessage(
      intl,
      'validations.notMatching.phoneNumber',
      'Phone numbers do not match',
    );
  } else if (newMobileNumberCountryCode !== confirmNewMobileNumberCountryCode) {
    errors.confirmNewMobileNumberCountryCode = formatMessage(
      intl,
      'validations.notMatching.phoneCode',
      'Country code does not match',
    );
  }

  return errors;
};

export default validate;
