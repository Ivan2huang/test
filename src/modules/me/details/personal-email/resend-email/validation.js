import { required, email } from '../../../../../helpers/validation';
import { formatMessage } from '../../../../../helpers/helpers';

const validate = (values, { intl }) => {
  const errors = {};
  const errorMessage = formatMessage(
    intl,
    'validations.invalid.email',
    'Enter valid email address',
  );

  if (required(values.newEmail) || email(values.newEmail)) {
    errors.newEmail = errorMessage;
  }

  return errors;
};

export default validate;
