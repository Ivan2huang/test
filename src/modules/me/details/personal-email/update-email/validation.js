import { required, email } from '../../../../../helpers/validation';
import { formatMessage } from '../../../../../helpers/helpers';

const validate = (values, { intl }) => {
  const errors = {};
  const errorMessage = formatMessage(
    intl,
    'validations.invalid.email',
    'Enter valid email address',
  );
  if (required(values.newEmail) || email(values.newEmail)) {
    errors.newEmail = errorMessage;
  } else if (required(values.confirmEmail) || email(values.confirmEmail)) {
    errors.confirmEmail = errorMessage;
  } else if (values.newEmail !== values.confirmEmail) {
    errors.confirmEmail = formatMessage(
      intl,
      'validations.notMatching.email',
      'Emails do not match',
    );
  }

  return errors;
};

export default validate;
