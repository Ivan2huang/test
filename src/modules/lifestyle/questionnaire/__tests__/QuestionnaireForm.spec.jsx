/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import QuestionnaireForm from '../QuestionnaireForm';
import withRedux from '../../../../redux/withReduxProvider';
import withIntl from '../../../../i18n/withIntlProvider';
import withTheme from '../../../../themes/withThemeProvider';
import { validate } from '../validation';

jest.mock('../validation', () => ({
  validate: jest.fn(() => ({})),
}));

jest.mock('../AboutMeContainer', () => props => {
  // eslint-disable-next-line global-require
  const { Field } = require('redux-form');
  return (
    <div>
      AboutMeContainer Component
      <span data-id="height" />
      <Field component="input" name="height" />
      <span>{mockPropsCapture(props)}</span>
    </div>
  );
});

jest.mock('../QuestionsSection', () => props => (
  <div>
    QuestionsSection Component
    <span>{mockPropsCapture(props)}</span>
  </div>
));

jest.mock('../future-me', () => ({
  FutureMeContainer: props => (
    <div>
      FutureMeContainer Component
      <span>{mockPropsCapture(props)}</span>
    </div>
  ),
}));

jest.mock('../../../../helpers/helpers', () => ({
  formatMessage: (intl, id, defaultMessage) => defaultMessage,
}));

jest.mock('../../../../uiComponents/Typography', () => props => (
  <div>
    Typography Component
    <span>{mockPropsCapture(props)}</span>
  </div>
));

describe('QuestionnaireForm Component', () => {
  const props = {
    choicesQuestions: [
      {
        questionOrder: '1',
        questionId: 'choices_2.1',
        dependentQuestion: false,
        name: 'smokingBehaviour',
        type: 'radio',
        text:
          'What is your current cigarettes (including hand-rolled cigarettes) smoking behaviour?',
        section: 'choices',
        options: [
          {
            label: 'Non-smoker',
            value: 'NonSmoker',
          },
          {
            label: 'Ex-smoker',
            value: 'ExSmoker',
          },
        ],
      },
      {
        questionOrder: '2',
        questionId: 'choices_2.2',
        dependentQuestion: false,
        name: 'alcoholConsumptionFrequency',
        type: 'radio',
        text:
          'How often you have a drink containing alcohol? 12.5 grams of alcohol per drink - Beer - 12oz (355ml), Wine - 5 oz (148ml), Spirits - 1.5 oz (44ml).',
        section: 'choices',
        options: [
          {
            label: 'Never',
            value: 'Never',
          },
          {
            label: 'Monthly or less',
            value: 'MonthlyOrLess',
          },
        ],
      },
    ],
    healthQuestions: [
      {
        questionOrder: '1',
        questionId: 'health_3.1',
        dependentQuestion: false,
        name: 'hereditaryDiabetes',
        type: 'radio',
        text:
          'Have any of your parents or siblings been diagnosed with diabetes?',
        options: [
          {
            label: 'Yes',
            value: 'Yes',
          },
          {
            label: 'No',
            value: 'No',
          },
        ],
      },
      {
        questionOrder: '2',
        questionId: 'health_3.2',
        dependentQuestion: false,
        name: 'highBloodPressure',
        type: 'radio',
        text:
          'Has a doctor told you previously that youâ€™ve got high blood pressure or are you on medication for high blood pressure?',
        options: [
          {
            label: 'Yes',
            value: 'Yes',
          },
          {
            label: 'No',
            value: 'No',
          },
        ],
      },
    ],
    submitQuestionnaire: jest.fn(),
    change: jest.fn(),
    futureMeImageChanged: true,
    faceAgingImageError: false,
    toggleErrorModal: jest.fn(),
  };
  const setUp = (componentProps = props) => {
    const Component = withIntl(withTheme(withRedux(QuestionnaireForm)));
    return render(<Component {...componentProps} />);
  };

  afterEach(() => {
    props.submitQuestionnaire.mockReset();
  });

  it('should match snapshot', () => {
    const { container } = setUp();

    expect(container).toMatchSnapshot();
  });

  it('should submit the questionnaire form', () => {
    const newProps = {
      ...props,
      initialValues: {
        aboutMe: {
          height: 150,
          weight: 75,
          waistCircumference: 100,
          isAsian: true,
        },
        choices: {
          smokingBehaviour: 'NonSmoker',
          alcoholConsumptionFrequency: 'Never',
          alcoholConsumptionAmount: 'OneOrTwo',
        },
        health: {
          hereditaryDiabetes: 'Yes',
          highBloodPressure: 'No',
          sleepDuration: 'Long',
        },
        futureMe: {
          image: 'test.png',
        },
      },
    };
    const formData = {
      aboutMe: {
        height: 150,
        weight: 75,
        waistCircumference: 100,
        isAsian: true,
      },
      choices: {
        smokingBehaviour: 'NonSmoker',
        alcoholConsumptionFrequency: 'Never',
      },
      health: {
        hereditaryDiabetes: 'Yes',
        highBloodPressure: 'No',
      },
      futureMe: {
        image: 'test.png',
      },
      isFaceAgingImageChanged: true,
    };

    const { getByTestId } = setUp(newProps);
    const submitButton = getByTestId('btn-submit-questionnaire');
    fireEvent.click(submitButton);

    expect(props.submitQuestionnaire).toHaveBeenCalledWith(formData);
  });

  it('should scroll to first validation error', () => {
    validate.mockReturnValue({
      aboutMe: {
        height: {
          _error: true,
        },
      },
      choices: {},
    });

    const { container, getByTestId } = setUp();
    const firstErrorElement = container.querySelector('[data-id=height]');
    firstErrorElement.scrollIntoView = jest.fn();

    const submitButton = getByTestId('btn-submit-questionnaire');
    fireEvent.click(submitButton);

    expect(firstErrorElement.scrollIntoView).toHaveBeenCalledWith(true);
  });
});
