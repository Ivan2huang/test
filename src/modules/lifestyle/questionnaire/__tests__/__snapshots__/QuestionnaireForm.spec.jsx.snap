// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireForm Component should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root ReduxForm-cardRoot-1 MuiPaper-rounded"
  >
    <form
      autocomplete="off"
    >
      <div>
        AboutMeContainer Component
        <span
          data-id="height"
        />
        <input
          name="aboutMe.height"
          value=""
        />
        <span>
          fieldChange = function fieldChange
        </span>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div>
        QuestionsSection Component
        <span>
          title = My choices
          imageUrl = /static/questionnaire_my_choices.svg
          questions = [object Object],[object Object]
        </span>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div>
        QuestionsSection Component
        <span>
          title = My health
          imageUrl = /static/questionnaire_my_health.svg
          questions = [object Object],[object Object]
        </span>
      </div>
      <div>
        FutureMeContainer Component
        <span>
          fieldChange = function fieldChange
        </span>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiBox-root MuiBox-root-37"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="btn-submit-questionnaire"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Show results
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
  <div
    class="MuiBox-root MuiBox-root-97"
  >
    <div>
      Typography Component
      <span>
        type = style-8
        color = highEmphasis
        children = Disclaimer
      </span>
    </div>
    <div>
      Typography Component
      <span>
        type = style-8
        color = mediumEmphasis
        children = This questionnaire is designed to provide you with an indication of your lifestyle and its impact on your health/wellness. It is NOT intended to constitute professional medical advice, diagnosis or treatment, and is NOT a substitute for a medical health assessment by a qualified medical professional. If youâ€™re experiencing signs and symptoms we recommend you seek medical attention from a qualified medical professional. By submitting this questionnaire, you do so in at your own choice and in agreement with this disclaimer.
      </span>
    </div>
  </div>
</div>
`;
